app/org/sagebionetworks/bridge/dynamodb/DynamoUserConsentDao.java

```diff
@@ -30,32 +30,26 @@
 @Component
 public class DynamoUserConsentDao implements UserConsentDao {
 
-    private DynamoDBMapper mapper2;
     private DynamoDBMapper mapper;
 
-    @Resource(name = "userConsentDdbMapper2")
-    public final void setDdbMapper2(DynamoDBMapper mapper2) {
-        this.mapper2 = mapper2;
-    }
-    
     @Resource(name = "userConsentDdbMapper")
     public final void setDdbMapper(DynamoDBMapper mapper) {
         this.mapper = mapper;
     }
 
     @Override
-    public UserConsent giveConsent(String healthCode, StudyConsent studyConsent, long signedOn) {
+    public UserConsent giveConsent(String healthCode, SubpopulationGuid subpopGuid, long consentCreatedOn, long signedOn) {
         checkArgument(isNotBlank(healthCode));
-        checkNotNull(studyConsent);
+        checkNotNull(subpopGuid);
         checkArgument(signedOn > 0L);
 
-        UserConsent activeConsent = getActiveUserConsent(healthCode, new StudyIdentifierImpl(studyConsent.getStudyKey()));
-        if (activeConsent != null && activeConsent.getConsentCreatedOn() == studyConsent.getCreatedOn()) {
-            throw new BridgeServiceException("Consent already exists.", HttpStatus.SC_CONFLICT);
+        UserConsent activeConsent = getActiveUserConsent(healthCode, subpopGuid);
+        if (activeConsent != null && activeConsent.getConsentCreatedOn() == consentCreatedOn) {
+            throw new EntityAlreadyExistsException(activeConsent);
```

Comment:
As per IM conversation, we don't currently expose the UserConsent anywhere. I'm okay with exposing the UserConsent, but if we do, we'll need to do the work to make sure it serializes correctly and that it doesn't expose certain fields to the user, most notably healthCode and studyId